orca-retrofit/src/main/groovy/com/netflix/spinnaker/orca/retrofit/exceptions/RetrofitExceptionHandler.groovy

```diff
@@ -66,14 +66,25 @@ class RetrofitExceptionHandler implements ExceptionHandler {
       responseDetails.kind = e.kind
       responseDetails.status = properties.status ?: null
       responseDetails.url = properties.url ?: null
-      boolean shouldRetry = !isMalformedRequest(e) &&
-        (isNetworkError(e) || isGatewayTimeout(e) || isThrottle(e)) &&
-        isIdempotentRequest(e)
-      return new ExceptionHandler.Response(e.class.simpleName, stepName, responseDetails, shouldRetry)
+
+      return new ExceptionHandler.Response(e.class.simpleName, stepName, responseDetails, shouldRetry(e))
+    }
+  }
+
+  boolean shouldRetry(RetrofitError e) {
+    if (isMalformedRequest(e)) {
+      return false
     }
+
+    // retry on 503 even for non-idempotent requests
+    if (e.kind == HTTP && e.response.status == HTTP_UNAVAILABLE) {
+      return true
+    }
+
+    return isIdempotentRequest(e) && (isNetworkError(e) || isGatewayErrorCode(e) || isThrottle(e))
   }
 
-  boolean isGatewayTimeout(RetrofitError e) {
+  boolean isGatewayErrorCode(RetrofitError e) {
     e.kind == HTTP && e.response.status in [HTTP_BAD_GATEWAY, HTTP_UNAVAILABLE, HTTP_GATEWAY_TIMEOUT]
```

Comment:
Wouldn't it make sense to remove `HTTP_UNAVAILABLE` from this list since you've handled it already?

Reply:
We could do that, but on the other hand if we keep it like that the method does more what it says -- it does tell you if you're having a gateway error of any kind. That opens it to future reuse in a different context potentially